source/docs/software/basic-programming/java-units.rst

```diff
@@ -1,58 +1,107 @@
+
+.. code-block:: java
+
+  Measure<Distance> wheelDiameter = Inches.of(6);
 
-New ``Measure`` objects can be created by calling the ``Unit#of`` method on the appropriate unit object, such as ``Inches.of(8)`` or ``Volts.of(13.5)``.
-``Measure`` is a generic type that contains the type of unit the measurement uses, such as ``Angle`` or ``Distance``. For example, ``Measure<Distance> wheelDiameter = Inches.of(6)`` is a measurement object corresponding to a distance of six inches, and ``Measure<Velocity<Angle>> maxRPM = RPM.of(5640)`` is a measurement corresponding to an angular velocity of 5,640 RPM.
+Other measures can also be created using their ``Unit.of`` method:
 
 .. code-block:: java
 
-   import edu.wpi.first.units.Distance;
-   import edu.wpi.first.units.Measure;
-   import static edu.wpi.first.units.Units.*;
+  Measure<Mass> kArmMass = Kilograms.of(1.423);
+  Measure<Distance> kArmLength = Inches.of(32.25);
+  Measure<Angle> kMinArmAngle = Degrees.of(5);
+  Measure<Angle> kArmMaxTravel = Rotations.of(0.45);
+  Measure<Velocity<Distance> kMaxSpeed = MetersPerSecond.of(2.5);
 
-   public class Constants {
-     public static class DriveConstants {
-       /** The distance between the front and rear axles. **/
-       public static final Measure<Distance> kWheelBase = Inches.of(28);
+Performing Calculations
+~~~~~~~~~~~~~~~~~~~~~~~
 
-       /** The distance between left and right wheels. **/
-       public static final Measure<Distance> kTrackWidth = Inches.of(26);
-     }
-   }
+The ``Measure`` class also supports arithmetic operations, such as addition, subtraction, multiplication, and division. These are done by calling methods on the objects. These operations always ensure that the units of measurement are compatible before performing the calculation, and they return a new ``Measure`` object. For example, you can add two ``Measure<Distance>`` objects together, even if they have different units:
+
+.. code-block:: java
+
+  Measure<Distance> distance1 = Inches.of(10);
+  Measure<Distance> distance2 = Meters.of(0.254);
 
-Because the ``Measure`` objects are Java objects, arithmetic must be done by calling methods on the objects. Note that each operation will return a new ``Measure`` object.
+  Measure<Distance> totalDistance = distance1.plus(distance2);
+
+This code will automatically convert the units of measurement to the same unit before adding the two distances. The resulting ``totalDistance`` object will be a new ``Measure<Distance>`` object that has a value of 0.508 meters, or 20 inches.
```

Comment:
This is a useful explanation, but I think it should be mentioned that the unit conversion is entirely internal and the whole point of the library is to abstract this away. Maybe you could preface it with "Internally, this works by" or something?